src/app/vendors/hooks.ts

```diff
@@ -0,0 +1,36 @@
+import { DestinyAccount } from 'app/accounts/destiny-account';
+import { loadingTracker } from 'app/shell/loading-tracker';
+import { refresh$ } from 'app/shell/refresh-events';
+import { useThunkDispatch } from 'app/store/thunk-dispatch';
+import { useEventBusListener } from 'app/utils/hooks';
+import { useCallback, useEffect } from 'react';
+import { loadAllVendors } from './actions';
+
+/**
+ * Loads all vendors for the given account+character, and tries refreshing them
+ * when the app refreshes.
+ */
+export function useLoadVendors(
+  account: DestinyAccount,
+  storeId: string | undefined,
+  active = true
+) {
+  const dispatch = useThunkDispatch();
+  useEffect(() => {
+    if (storeId && active) {
+      dispatch(loadAllVendors(account, storeId));
```

Comment:
Should this also be wrapped in the loading tracker?

Reply:
The original code didn't do that, and it's probably because it's meant for the initial vendor load where the Vendors screen already shows a full-screen DIM loading animation, and the LO toggle also gets a spinning loading icon for the initial load.

Reply:
Got it